### PR TITLE
Dispatch audit event when changing ACL rules

### PR DESCRIPTION
**Pull request by CarlSchwan**
Implements #1744

### CODE REVIEW

File: lib/ACL/RuleManager.php [hunk 1]
```diff
@@ -293,6 +295,20 @@ public function saveRule(Rule $rule): void {
 				->andWhere($query->expr()->eq('mapping_type', $query->createNamedParameter($rule->getUserMapping()->getType())))
 				->andWhere($query->expr()->eq('mapping_id', $query->createNamedParameter($rule->getUserMapping()->getId())));
 			$query->executeStatement();
+
+			if ($rule->getUserMapping()->getType() === 'user') {
+				$logMessage = 'The ACL rule was updated for file/folder with id: "%s" for the user "%s" with permission "%s" and mask "%s"';
```
Comment (PVince81) [inline lib/ACL/RuleManager.php:300]: Maybe change to this so we group the "to" and "for" together
"The ACL rule was updated to permission "%s" and mask "%s" for file/folder with id %s for user "%s""

File: lib/ACL/RuleManager.php [hunk 2]
```diff
@@ -293,6 +295,20 @@ public function saveRule(Rule $rule): void {
...
+			} else {
+				$logMessage = 'The ACL rule was updated for file/folder with id: "%s" for the group "%s" with permission "%s" and mask "%s"';
+			}
+			$params = [
+				'fileId' => $rule->getFileId(),
```
Comment (PVince81) [inline lib/ACL/RuleManager.php:305]: are we mixing parameter order with named parameters ? with associative arrays I'm not sure if the order is guaranteed
is this how we do it in other places, looks weird

File: lib/ACL/RuleManager.php [hunk 3]
```diff
@@ -293,6 +295,20 @@ public function saveRule(Rule $rule): void {
...
+				'user' => $rule->getUserMapping()->getDisplayName() . '(' . $rule->getUserMapping()->getId() . ')',
```
Comment (PVince81) [inline lib/ACL/RuleManager.php:306]: getUserMapping() also for groups ?

File: lib/ACL/RuleManager.php [hunk 4]
```diff
@@ -293,6 +295,26 @@ public function saveRule(Rule $rule): void {
 				->andWhere($query->expr()->eq('mapping_type', $query->createNamedParameter($rule->getUserMapping()->getType())))
 				->andWhere($query->expr()->eq('mapping_id', $query->createNamedParameter($rule->getUserMapping()->getId())));
 			$query->executeStatement();
+
+			if ($rule->getUserMapping()->getType() === 'user') {
+				$logMessage = 'The ACL rule was updated to permission "%s" ans mask "%s" for file/folder with id: "%s" for user "%s"';
```
Comment (PVince81) [inline lib/ACL/RuleManager.php:300]: s/ans/and/ all messages

I'd also remove the colon to make it flow better

**Review by PVince81 — APPROVED** [review]
:+1:

would be good to also add a unit test that catches the event and checks at least the params, I think we did this in other places

File: lib/ACL/RuleManager.php (hunk 4 above)
Comment (CarlSchwan) [reply lib/ACL/RuleManager.php]: done

**Pull request merged**